Feature/automl (#199)
nils-braun/dask-sql: docs/pages/machine_learning.rst

    -- DESCRIBE MODEL <model_name>.
    DESCRIBE MODEL my_model

5. Hyperparameter Tuning
-------------------------

> Wants to increase the performance of your model by tuning the

Reviewer: ```suggestion
Want to increase the performance of your model by tuning the
```

> parameters ? Use the Hyperparameter tuning directly

Reviewer: ```suggestion
parameters? Use hyperparameter tuning directly
```

in SQL using below SQL syntax, choose different tuners

> from dask_ml package based on memory and compute constraints and

Reviewer: ```suggestion
from the dask_ml package based on memory and compute constraints and
```

> for more details refer  `dask ml documentation <https://ml.dask.org/hyper-parameter-search.html#incremental-hyperparameter-optimization>`_

Reviewer: ```suggestion
for more details refer  to the `dask ml documentation <https://ml.dask.org/hyper-parameter-search.html#incremental-hyperparameter-optimization>`_
```

.. code-block:: sql

>  CREATE EXPERIMENT my_exp WITH(

Reviewer: ```suggestion
 CREATE EXPERIMENT my_exp WITH (
```

    model_class = 'sklearn.ensemble.GradientBoostingClassifier',
    experiment_class = 'dask_ml.model_selection.GridSearchCV',
    tune_parameters = (n_estimators = ARRAY [16, 32, 2],
                    learning_rate = ARRAY [0.1,0.01,0.001],
                   max_depth = ARRAY [3,4,5,10]
                   ),
    target_column = 'target'
    ) AS (
        SELECT x, y, x*y > 0 AS target
        FROM timeseries
        LIMIT 100
    )

5.1 Automl in SQL
-----------------

> Wants to try different models with different parameters in SQL ?, Now you can

Reviewer: ```suggestion
Want to try different models with different parameters in SQL? Now you can
```

> start automl experiment with the help ``tpot`` framework which trains

Reviewer: ```suggestion
start automl experiments with the help of the ``tpot`` framework which trains
```

> and evaluates number of different sklearn compatible models and uses dask for

Reviewer: ```suggestion
and evaluates a number of different sklearn compatible models and uses dask for
```

distributing the work across the dask clusters.

> use below SQL syntax for automl and for more details refer

Reviewer: ```suggestion
Use below SQL syntax for automl and for more details refer to the
```

`tpot automl framework <https://epistasislab.github.io/tpot/>`_

.. code-block:: sql

    CREATE EXPERIMENT my_exp WITH (
            automl_class = 'tpot.TPOTClassifier',
            automl_kwargs = (population_size = 2 ,
            generations=2,
            cv=2,
            n_jobs=-1,
            use_dask=True,
            max_eval_time_mins=1),
            target_column = 'target'
            ) AS (
                SELECT x, y, x*y > 0 AS target
                FROM timeseries
                LIMIT 100
            )

After the experiment was completed, both hyperparameter tuner and

> automl experiments stores the best model of the experiment in sql context

Reviewer: ```suggestion
automl experiments stores the best model of the experiment in the sql context
```
